domain/ide/command/ignores_request.go

```diff
@@ -0,0 +1,289 @@
+		return fmt.Errorf("no data returned from ignore workflow")
+	}
+
+	output, ok := response[0].GetPayload().([]byte)
+	if !ok {
+		return fmt.Errorf("invalid response from ignore workflow")
+	}
+
+	err = updateIssueWithIgnoreDetails(cmd.c, output, issue)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *submitIgnoreRequest) createCreateConfiguration(engine workflow.Engine, findingsId string, contentRoot types.FilePath) (configuration.Configuration, error) {
+	if len(cmd.command.Arguments) < 5 {
+		return nil, fmt.Errorf("insufficient arguments for ignore-create workflow")
+	}
+
+	ignoreType, ok := cmd.command.Arguments[2].(string)
+	if !ok {
+		return nil, fmt.Errorf("ignoreType should be a string")
+	}
+	reason, ok := cmd.command.Arguments[3].(string)
+	if !ok {
+		return nil, fmt.Errorf("reason should be a string")
+	}
+	expiration, ok := cmd.command.Arguments[4].(string)
+	if !ok {
+		return nil, fmt.Errorf("expiration should be a string")
+	}
+
+	gafConfig := engine.GetConfiguration().Clone()
+	gafConfig.Set(ignore_workflow.FindingsIdKey, findingsId)
+	gafConfig.Set(ignore_workflow.IgnoreTypeKey, ignoreType)
+	gafConfig.Set(ignore_workflow.ReasonKey, reason)
+	gafConfig.Set(ignore_workflow.ExpirationKey, expiration)
+	gafConfig.Set(ignore_workflow.EnrichResponseKey, true)
+	gafConfig.Set(ignore_workflow.InteractiveKey, false)
+	gafConfig.Set(configuration.INPUT_DIRECTORY, contentRoot)
+	return gafConfig, nil
+}
+
+func (cmd *submitIgnoreRequest) editIgnoreRequest(engine workflow.Engine, findingsId string, contentRoot types.FilePath, issue types.Issue) error {
+	gafConfig, err := cmd.createEditConfigurations(engine, findingsId, contentRoot)
+	if err != nil {
+		return err
+	}
+
+	response, err := engine.InvokeWithConfig(ignore_workflow.WORKFLOWID_IGNORE_CREATE, gafConfig)
+	if err != nil {
+		return err
+	}
+
+	if len(response) == 0 {
+		return fmt.Errorf("no data returned from ignore workflow")
+	}
+
+	output, ok := response[0].GetPayload().([]byte)
+	if !ok {
+		return fmt.Errorf("invalid response from ignore workflow")
+	}
+
+	err = updateIssueWithIgnoreDetails(cmd.c, output, issue)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *submitIgnoreRequest) createEditConfigurations(engine workflow.Engine, findingsId string, contentRoot types.FilePath) (configuration.Configuration, error) {
+	if len(cmd.command.Arguments) < 5 {
+		return nil, fmt.Errorf("insufficient arguments for ignore-edit workflow")
+	}
+
+	ignoreType, ok := cmd.command.Arguments[2].(string)
+	if !ok {
+		return nil, fmt.Errorf("ignoreType should be a string")
+	}
+	reason, ok := cmd.command.Arguments[3].(string)
+	if !ok {
+		return nil, fmt.Errorf("reason should be a string")
+	}
+	expiration, ok := cmd.command.Arguments[4].(string)
+	if !ok {
+		return nil, fmt.Errorf("expiration should be a string")
+	}
+	ignoreId, ok := cmd.command.Arguments[5].(string)
+	if !ok {
+		return nil, fmt.Errorf("ignoreId should be a string")
+	}
```

Comment:
some of the argument checks are also duplications, please refactor accordingly to only have the necessary code once.